bsonutil/bsonutil.go

```diff
@@ -427,3 +427,43 @@ func parseNumberLongField(jsonValue interface{}) (int64, error) {
 		return 0, errors.New("expected $numberLong field to have string value")
 	}
 }
+
+func Bson2Float64(data interface{}) (float64, bool) {
```

Comment:
Yea, I know other types may not be possible for the index values. Since this function is placed in bsonutil, I want it to be generic instead of only serve the purpose of convertLegacyIndexes. Maybe in that case, we can keep those types?

Reply:
BSON only has 4 numeric types. So the other types probably won't be used, even outside of indexes.

Reply:
ah, I see. OK. I can remove the other types if BSON doesn't support them.

Reply:
It's possible people unmarshal into non-standard types. But the 4 types listed above are the standard.